File: algorithms/804/804-unique-morse-code-words.py
def uniqueMorseRepresentations(words) -> int:
    ms = [".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--",
          "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."]
    tem = []

    for i in range(len(words)):
        s = ''
        for j in words[i]:
            s += ms[ord(j)-97]
        tem.append(s)

    return len(set(tem))


if __name__ == "__main__":
    print(uniqueMorseRepresentations(["gin", "zen", "gig", "msg"]))

File: algorithms/804/804-unique-morse-code-words.js
/**
 * @param {string[]} words
 * @return {number}
 */
var uniqueMorseRepresentations = function (words) {
    let morseWords = ['.-', '-...', '-.-.', '-..', '.', '..-.', '--.', '....', '..', '.---', '-.-', '.-..', '--', '-.', '---', '.--.', '--.-', '.-.', '...', '-', '..-', '...-', '.--', '-..-', '-.--', '--..'];

    return new Set(words.map(word => {
        return Array.from(word).reduce((total, w) => {
            return total += morseWords[w.charCodeAt() - 97]
        }, '')
    })).size
};

console.log(uniqueMorseRepresentations(["gin", "zen", "gig", "msg"]));